### PR TITLE
paper for JOSS

### PR DESCRIPTION
**Pull request by munkm**
This is my first stab at the JOSS paper. I haven't done any performance comparison yet, and the equations need to be formatted a bit, but this is the general gist. Is there anything that's missing that we need to cover?

### CODE REVIEW

File: paper/paper.bib [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	pages = {90},
+	file = {Snapshot:/Users/madicken/Zotero/storage/5IJM9CLA/MCSE.2007.html:text/html}
+}
+
+@article{turk_yt:_2010,
+	title = {yt: {A} multi-code analysis toolkit for astrophysical simulation data},
+	volume = {192},
+	shorttitle = {yt},
+	number = {1},
+	journal = {The Astrophysical Journal Supplement Series},
```
Comment (matthewturk) [inline paper/paper.bib]: I *think* the publication date is actually 2011, weirdly.

File: paper/paper.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+$$ T_{client} < T_{server}
+n*t_{imaage calc, client} + t_{pull, data} < n*[t_{image_calc, server} + \ 
+t_{pull, image}]
+$$
+
+The time to pull an image or data is dependent on the data size and the
+transfer rate. 
+$T_{client}$ will be lower than $T_{server}$ as the number of
+interactions $n$ grows, as the size of the image (data$_{image}$) grows, and as
+the time to calculate the imagae on the client $t_{image calc, client}$
```
Comment (matthewturk) [inline paper/paper.md]:
```suggestion
the time to calculate the image on the client $t_{image calc, client}$
```

File: paper/paper.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+sparse regions will be more efficient to transfer to the client and subsequently
+calculate and pixelize there. Pixelizing a dataset with large, sparse regions of low
+resolution, such as one caclated on an adaptive mesh, 
+with a fixed higher resolution will require recaculating and sending
+pixel values for a region that may only be represented by a single value. 
+
+# The WebAssembly backend
+
+To allow for efficient data loading into the browser we chose to use Rust
+compiled to WebAssembly. The WebAssembly backing of widgyts allows for binary
```
Comment (matthewturk) [inline paper/paper.md]:
```suggestion
compiled to WebAssembly. The WebAssembly backing of widgyts allows for binary, zero-copy
```

File: paper/paper.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+storage of loaded data on the client side, and WebAssembly has been designed to
+interface well with JavaScript. Further, the primitive structure of WebAssembly
+reduces the time to calculate the image in the browser. Finally, WebAssembly
+is executed in a sandboxed environment separate from other processes 
+and is memory safe. 
+
+While yt can access data at an arbitrary location within the dataset, widgyts
+is structured to access any data within a 2D slice. Thus, only a slice of the
+data is uploaded client-side, not the entire dataset. For the large, sparse
+datasets that widgyts has been designed for, it would be unfeasible to upload
```
Comment (matthewturk) [inline paper/paper.md]:
```suggestion
datasets that widgyts has been designed for, it would be infeasible to upload
```

File: paper/paper.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+the entire dataset into the browser. However, a new slice in
+the third dimension will require a new data upload from the server so not all
+exploration of the dataset can be performed exclusively client-side.
+
+todo: performance comparison with datasets of varying sparsity and show how
```
Comment (matthewturk) [inline paper/paper.md]: I don't think this is super necessary.

File: paper/paper.bib (hunk 1 above)
Comment (munkm) [reply paper/paper.bib]: Yep! It looks like Zotero scraped the draft date not the publication date.

**Pull request merged**